File: TestBank/app/src/main/java/monteoliva/testbank/view/components/Progress.java
package monteoliva.testbank.view.components;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;

import monteoliva.testbank.R;

public class Progress extends LinearLayout {
	/**
	 * Constructor
	 *
	 * @param context
	 */
	public Progress(@NonNull Context context) { super(context); init(context); }
	public Progress(@NonNull Context context, @NonNull AttributeSet attrs) { super(context, attrs); init(context);}

	/**
	 * Metodo de inicializacao
	 *
	 * @param context
	 */
	private void init(Context context) {
		setBackgroundColor(ContextCompat.getColor(context, android.R.color.transparent));

		final LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

		inflater.inflate(R.layout.progress, this);
	}

	public void hide() { setVisibility(View.GONE);    }
	public void show() { setVisibility(View.VISIBLE); }
}

File: app/src/main/java/br/com/testeandroidv2/view/components/Progress.kt
package br.com.testeandroidv2.view.components

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import androidx.core.content.ContextCompat

import br.com.testeandroidv2.R

class Progress(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs) {
    lateinit var view: View

    init {
        start(context, attrs)
    }

    private fun start(context: Context, attrs: AttributeSet) {
        // seta o Background
        setBackgroundColor(ContextCompat.getColor(context, android.R.color.transparent))

        // pega o inflater
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        // pega a View
        view = inflater.inflate(R.layout.progress, this)
    }

    fun show() { view.visibility = View.VISIBLE }
    fun hide() { view.visibility = View.GONE    }
}